File: projects/calculator/csharp/Calculator.cs
using System;

class Calculator {
    static void Main() {
        Console.WriteLine("Simple Calculator");
        Console.WriteLine("1. Add  2. Subtract  3. Multiply  4. Divide");
        Console.Write("Choice: ");
        string choice = Console.ReadLine();

        Console.Write("First number: ");
        double a = double.Parse(Console.ReadLine());
        Console.Write("Second number: ");
        double b = double.Parse(Console.ReadLine());

        switch (choice) {
            case "1": Console.WriteLine($"Result: {a + b}"); break;
            case "2": Console.WriteLine($"Result: {a - b}"); break;
            case "3": Console.WriteLine($"Result: {a * b}"); break;
            case "4":
                if (b != 0) Console.WriteLine($"Result: {a / b}");
                else Console.WriteLine("Cannot divide by zero");
                break;
            default: Console.WriteLine("Invalid"); break;
        }
    }
}

File: projects/calculator/typescript/calculator.ts
import * as readline from "readline";

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function ask(q: string): Promise<string> {
  return new Promise(resolve => rl.question(q, resolve));
}

async function main() {
  console.log("Simple Calculator");
  console.log("1. Add  2. Subtract  3. Multiply  4. Divide");

  const choice = await ask("Choice: ");
  const a = Number(await ask("First number: "));
  const b = Number(await ask("Second number: "));

  if (choice === "1") console.log("Result:", a + b);
  else if (choice === "2") console.log("Result:", a - b);
  else if (choice === "3") console.log("Result:", a * b);
  else if (choice === "4") {
    if (b !== 0) console.log("Result:", a / b);
    else console.log("Cannot divide by zero");
  } else console.log("Invalid");

  rl.close();
}

main();
